dags/challenge_dag_factory.py

```diff
@@ -0,0 +1,173 @@
+import os
+import uuid
+from datetime import datetime
+import yaml
+import pandas as pd
+
+from airflow.decorators import dag, task
+from airflow.models import Param
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+from orca.services.nextflowtower import NextflowTowerHook
+from orca.services.nextflowtower.models import LaunchInfo
+from orca.services.synapse import SynapseHook
+
+# Define the path to your challenge configuration file.
+CONFIG_FILE = os.path.join(os.path.dirname(__file__), "challenge_configs.yaml")
+
+
+def load_challenge_configs():
+    """Load challenge configurations from a YAML file."""
+    with open(CONFIG_FILE, "r") as f:
+        return yaml.safe_load(f)
```

Comment:
When we get this deployed out to the airflow server running in K8s this is going to be an area that could fail. I honestly do not know if the git cloning process downloads the files like this or keeps them in memory.

If this does end up becoming an issue we would change this to access the ["raw" github url](https://raw.githubusercontent.com/Sage-Bionetworks-Workflows/orca-recipes/2ea5cba6e0dcadb296e3330b626dfa10b541a934/dags/challenge_configs.yaml) and load the HTTP response into memory.

Reply:
> we would change this to access the ["raw" github url](https://raw.githubusercontent.com/Sage-Bionetworks-Workflows/orca-recipes/2ea5cba6e0dcadb296e3330b626dfa10b541a934/dags/challenge_configs.yaml) and load the HTTP response into memory

I went ahead and [made this change](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/pull/85/commits/580d61cedd083d3ed87207a377dc1cae108b64fc) preemptively so it's more robust (I have it pointed to this branch until this is ready for merge). thanks for the suggestion!